### ISSUE TITLE
Bugs for 3.2.1

### ISSUE THREAD

**Issue by poshul** [OpenMS/OpenMS#7633]
- [x] Issues with numpy? (merge numpy 2 support, test nightly)
- [x] Path separator bug KNIME (fixed)
- [x] Missing param in KNIME as well (ask Chris, if no good short term solution, revert at end of week)
- [ ] QT stuff? (ask Julianus)

**Comment by timosachsenberg** [comment]
@jpfeuffer Recently, we received another report on GitHub (https://github.com/Nesvilab/FragPipe/issues/1836) and Guo Ci figured out the reason (https://github.com/Nesvilab/FragPipe/issues/1543#issuecomment-2435555771).
It seems that if the user installed Python as an administrator but installed the Python packages as a non-administrator, PyOpenMS breaks.

From Dave: I have run into a problem at the intersection of OpenMS and FragPipe, and I believe I will need your help to solve it!  I believe it is similar to a problem first reported in February (https://github.com/Nesvilab/FragPipe/issues/1431).

I used FragPipe 21.1 to power our recent paper on DIA data analysis (https://doi.org/10.1007/s42485-024-00166-4), so I thought I could easily use the software for a new DIA set in the worm lab I've joined.  I decided to update to the newest FragPipe (v22) and support tools (EasyPQP 0.1.50) for that purpose.

The error itself looks like this:
```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Scripts\easypqp.exe\__main__.py", line 4, in <module>
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\site-packages\easypqp\main.py", line 9, in <module>
    from .convert import conversion, basename_spectralfile, conversion_psm, drop_psm_columns
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\site-packages\easypqp\convert.py", line 18, in <module>
    import pyopenms as po
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyopenms\__init__.py", line 109, in <module>
    raise e
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyopenms\__init__.py", line 67, in <module>
    from ._all_modules import *  # pylint: disable=wildcard-import; lgtm(py/polluting-import)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\site-packages\pyopenms\_all_modules.py", line 1, in <module>
    from ._pyopenms_1 import *  # pylint: disable=wildcard-import; lgtm(py/polluting-import)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
ImportError: DLL load failed while importing _pyopenms_1: A dynamic link library (DLL) initialization routine failed.
Traceback (most recent call last):
  File "H:\software\fragpipe-22.0\tools\speclib\gen_con_spec_lib.py", line 580, in <module>
    main()
  File "H:\software\fragpipe-22.0\tools\speclib\gen_con_spec_lib.py", line 552, in main
    main_easypqp(params, irt_df, allcmds, easypqp_convert_cmds)
  File "H:\software\fragpipe-22.0\tools\speclib\gen_con_spec_lib.py", line 306, in main_easypqp
    subprocess.run([os.fspath(params.easypqp), '--version'], check=True)
  File "C:\Users\dtabb\AppData\Local\Programs\Python\Python311\Lib\subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['C:\\Users\\dtabb\\AppData\\Local\\Programs\\Python\\Python311\\Scripts\\easypqp.exe', '--version']' returned non-zero exit status 1.

======================================================================
Error when loading pyOpenMS libraries!
```

**Comment by timosachsenberg** [comment]
@cbielow we struggle with the nan as default for score parameters in IDFilter OpenMS (breaks our KNIME nodes). Would you be fine with reverting it for now. Seems your change was not critical.

**Comment by jpfeuffer** [comment]
One solution to solve all python problems once and for all. Conda.
Pypi was never made for version compatibility of C libraries.

**Comment by jpfeuffer** [comment]
Also solves all Windows install problems magically for webapps etc.

**Comment by cbielow** [comment]
> @cbielow we struggle with the nan as default for score parameters in IDFilter OpenMS (breaks our KNIME nodes). Would you be fine with reverting it for now. Seems your change was not critical.

Taking the discussion on this here https://github.com/OpenMS/OpenMS/issues/7623

**Comment by timosachsenberg** [comment]
> One solution to solve all python problems once and for all. Conda.
> Pypi was never made for version compatibility of C libraries.

Ok but not sure if easypqp and others just switch to conda. Any hint how to debug this? Worked in last version apparently. Or could this also be numpy related?

**Comment by jpfeuffer** [comment]
I guess by doing the same things as in our other issue.
Figuring out what packages fragpipe loads, reproducing it in a clean python environment and reducing the imported libraries to the minimal set that crashes. If the only dependencies that those libraries share are numpy, then yes , it's probably numpy.
Other things to check are dependency walker, dlltracer etc.

**Comment by poshul** [comment]
Circling back around, do we have any more QT issues that need to be resolved?

**Comment by anesvi** [comment]
Hello, is there any update on this issue. We just had a hands-on FragPipe tutorial at a course in Barcelona and, for many students, EasyPQP crashed due to the Python/OpenMS issue. Are there any plans to fix it? Thank you, Alexey Nesvizhskii

**Comment by anesvi** [comment]
I am not sure why the issue was closed?

**Comment by timosachsenberg** [comment]
@anesvi we want to fix it asap. I guess this was a misunderstanding, as the original issue(s) https://github.com/Nesvilab/FragPipe/issues/1836 got closed. I can reopen it on our side. Sorry for the inconveniences.

**Comment by fcyu** [comment]
Hi @timosachsenberg ,

I apologize for the confusion. 

https://github.com/Nesvilab/FragPipe/issues/1836 was closed by the user, I guess it was because they thought our workaround approach resolved their issue. 

But this issue happens to many users recently and we have to let they reinstall the Python according to our workaround approach (https://github.com/Nesvilab/FragPipe/issues/1836#issuecomment-2430217130), which is a little annoying to most users.

Thanks for taking another look. And looking forward to the solutions.

Best,

Fengchao

**Comment by timosachsenberg** [comment]
@fcyu can you point me to the part in FragPipe that calls easypqp? is it here: https://github.com/Nesvilab/FragPipe/blob/7d3d8145839ee7ffe41aa39b633c3bee49d6eef8/MSFragger-GUI/tools/speclib/gen_con_spec_lib.py#L284-L334

**Comment by fcyu** [comment]
@timosachsenberg That's correct.

Thanks,

Fengchao

**Comment by poshul** [comment]
https://github.com/OpenMS/OpenMS/pull/7699 fixes the issue, I'll roll that out to 3.2 tomorrow.

**Comment by poshul** [comment]
@fcyu we've just rolled out 3.2.0.1 which fixes the issue, as well as opened a bug report with pyarrow since the issue was caused by an out-of-date C++ redistributable that is shipped there. Please let your users know, and feel free to reach out with any future problems.

Cheers,
Sam
